### PR TITLE
Another multipart example

### PR DESCRIPTION
**Pull request by kantimam**
pretty basic multipart form example. Mostly based of the multipart+s3 example just more basic and hopefully easier to understand for potatoes like me.

### CODE REVIEW

File: multipart-basic/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    println!("bytes={:#?}", form);
+    
+    println!("files={:#?}", files);
+    
+    Ok(HttpResponse::Ok().into())
+}
+
+fn index() -> HttpResponse {
+    let html = r#"<html>
```
Comment (robjtede) [inline multipart-basic/src/main.rs:20]: include the HTML from an external file using something like include_str! macro

File: multipart-basic/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    HttpResponse::Ok().body(html)
+}
+
+#[actix_rt::main]
+async fn main() -> std::io::Result<()> {
+    
+    std::env::set_var("RUST_LOG", "actix_server=info,actix_web=info");
+    std::fs::create_dir_all("./files").unwrap();
+
+    let ip = "0.0.0.0:3000";
```
Comment (robjtede) [inline multipart-basic/src/main.rs:67]: most examples start on `127.0.0.1:8080`

File: multipart-basic/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    HttpServer::new(|| {
+        App::new().wrap(middleware::Logger::default()).service(
+            web::resource("/")
+                .route(web::get().to(index))
```
Comment (robjtede) [inline multipart-basic/src/main.rs:72]: replace with simpler #[get] and #[post] macros on the handlers

File: multipart-basic/src/utils/payload_handler.rs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    while let Some(item) = payload.next().await {
+        let mut field: Field = item.expect(" split_payload err");
+        let content_type = field.content_disposition().unwrap();
+        let name = content_type.get_name().unwrap();
+        if name != "file" {
+            while let Some(chunk) = field.next().await {
+                let data = chunk.expect("split_payload err chunk");
+                /* convert bytes to string and print it  (just for testing) */
+                if let Ok(s)=str::from_utf8(&data){
+                    println!("{:?}", s);
```
Comment (robjtede) [inline multipart-basic/src/utils/payload_handler.rs:51]: if we're going to include debug prints, can we use log + env_logger and use debug! macro here

File: multipart-basic/src/utils/payload_handler.rs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    title: String,
+    description: String, 
+    count: u32
+}
+
+pub async fn split_payload(payload: &mut Multipart) -> (FormOut, Vec<UploadedFiles>) {
+    let mut files: Vec<UploadedFiles> = Vec::new();
+
+    /* fill with default values for now */
+    let mut form : FormOut=FormOut {
```
Comment (robjtede) [inline multipart-basic/src/utils/payload_handler.rs:36]: good practice to have an impl Default when you need to init like this

**Comment by JohnTitor** [comment]
My concern is that multiple examples may make us confused, like #295. And we have already three multipart examples. I'm not sure we really need this one, @actix/contributors thoughts?

File: multipart-basic/src/utils/payload_handler.rs [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct FormOut {
+    title: String,
+    description: String, 
+    count: u32
+}
+
+pub async fn split_payload(payload: &mut Multipart) -> (FormOut, Vec<UploadedFiles>) {
```
Comment (robjtede) [inline multipart-basic/src/utils/payload_handler.rs:32]: this function could use more comments imo

**Comment by robjtede** [comment]
On wether to include this example at all: actually hadnt spotted the existing basic multipart example. That one seems simpler than this even.

**Comment by kantimam** [comment]
ya fair enough. Kinda struggled to figure out how to get the text fields since both multipart examples focus on files mostly. Maybe add another input apart from only files to one of them?

**Comment by robjtede** [comment]
> Maybe add another input apart from only files to one of them?

Sounds good, if you want to add some text handling to that example a PR would be welcome IMO.

**Comment by kantimam** [comment]
oki :) will do that instead

**Comment by kantimam** [comment]
btw the actix-form-data example here https://crates.io/crates/actix-form-data is no longer valid? since i try to get it to work for like 2 days now.

**Comment by JohnTitor** [comment]
> btw the actix-form-data example here https://crates.io/crates/actix-form-data is no longer valid? since i try to get it to work for like 2 days now.

It isn't a part of the actix project and depends on actix-web v1.

**Pull request closed without merging**